examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,239 @@
+use std::time::Duration;
+
+use crossterm::event::{
+    self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, MouseEventKind,
+};
+use crossterm::ExecutableCommand;
+use ratatui::layout::{Constraint, Layout};
+use ratatui::style::{Modifier, Style};
+use ratatui::text::Line;
+use ratatui::widgets::{Block, Borders, Padding, Paragraph, Row, Table, TableState};
+use ratatui::Frame;
+
+static INFO_TEXT: &str = " (Esc/q) quit | (↑) move up | (↓) move down ";
+
+/// Application data model and state
+#[derive(Debug, Default)]
+struct Model {
+    table_items: Vec<Data>,
+    table_state: TableState,
+    running_state: RunningState,
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+enum RunningState {
+    #[default]
+    Running,
+    Done,
+}
+
+/// Data model
+#[derive(Debug, Default)]
+struct Data {
+    name: String,
+    address: String,
+    email: String,
+}
+
+#[derive(PartialEq, Copy, Clone)]
+enum Message {
+    SelectNext,
+    SelectPrevious,
+    Quit,
+}
+
+fn main() -> color_eyre::Result<()> {
+    tui::install_panic_hook();
+    let mut terminal = tui::init_terminal()?;
+    let mut model = Model {
+        table_items: generate_some_fans(),
+        ..Default::default()
+    };
+    stdout().execute(EnableMouseCapture)?;
+    while model.running_state != RunningState::Done {
+        // Render the current view
+        terminal.draw(|f| view(&mut model, f))?;
+
+        // Handle events and map to a Message
+        let mut current_msg = handle_event(&model)?;
+
+        // Process updates as long as they return a non-None message
+        while current_msg.is_some() {
+            current_msg = update(&mut model, current_msg.unwrap());
+        }
+    }
+    stdout().execute(DisableMouseCapture)?;
+    tui::restore_terminal()?;
+    Ok(())
+}
+
+fn view(model: &mut Model, frame: &mut Frame) {
+    let [top, bottom] = Layout::vertical([Constraint::Fill(1); 2]).areas(frame.area());
+
+    // Table setup
+    let header = Row::new(vec!["Name", "Address", "Email"])
+        .style(Style::default().add_modifier(Modifier::BOLD))
+        .height(1);
+
+    let rows = model.table_items.iter().map(|data| {
+        Row::new(vec![
+            data.name.as_str(),
+            data.address.as_str(),
+            data.email.as_str(),
+        ])
+        .height(2)
+    });
+
+    let table = Table::new(
+        rows,
+        [
+            Constraint::Length(20),
+            Constraint::Min(10),
+            Constraint::Fill(1),
+        ],
+    )
+    .header(header)
+    .row_highlight_style(Style::default().add_modifier(Modifier::REVERSED | Modifier::ITALIC))
+    .block(
+        Block::default()
+            .borders(Borders::ALL)
+            .title(" Terminal fans "),
+    );
+
+    frame.render_stateful_widget(table, top, &mut model.table_state);
+
+    // Select the first row if no row is selected
+    if model.table_state.selected().is_none() {
+        model.table_state.select_first();
+    }
```

Comment:
Does this need to go before the render call? Should it be in the main part of the app during some sort of update path rather than in the render code?

Reply:
Moved to main. Right after table init. Update only fires when there is a `current_msg`.